chain/x/oracle/handler.go

```diff
@@ -162,38 +162,38 @@ func handleMsgRequestData(ctx sdk.Context, k Keeper, m MsgRequestData, ibcData .
 		return nil, err
 	}
 
-	ctx.EventManager().EmitEvents(sdk.Events{sdk.NewEvent(
-		types.EventTypeRequest,
-		sdk.NewAttribute(types.AttributeKeyID, fmt.Sprintf("%d", id)),
-	)})
+	event := sdk.NewEvent(types.EventTypeRequest)
+	event = event.AppendAttributes(sdk.NewAttribute(types.AttributeKeyID, fmt.Sprintf("%d", id)))
+	for _, val := range validators {
+		event = event.AppendAttributes(sdk.NewAttribute(types.AttributeKeyValidator, val.String()))
+	}
+	ctx.EventManager().EmitEvent(event)
+
 	for _, raw := range env.GetRawRequests() {
 		err := k.PayDataSourceFee(ctx, raw.DataSourceID, m.Sender)
 		if err != nil { // We should fail here if the request tries to use an unknown data source.
 			return nil, err
 		}
-		ctx.EventManager().EmitEvents(sdk.Events{sdk.NewEvent(
+		// TODO: Consume more gas if using more raw requests.
+		ctx.EventManager().EmitEvent(sdk.NewEvent(
 			types.EventTypeRawRequest,
 			sdk.NewAttribute(types.AttributeKeyDataSourceID, fmt.Sprintf("%d", raw.DataSourceID)),
 			sdk.NewAttribute(types.AttributeKeyExternalID, fmt.Sprintf("%d", raw.ExternalID)),
 			sdk.NewAttribute(types.AttributeKeyCalldata, string(raw.Calldata)),
-		)})
+		))
 		// TODO: Remove raw request keeper. Make cacher and bandoracled parse from events.
 		err = k.AddRawRequest(ctx, id, raw)
 		if err != nil {
 			return nil, err
 		}
 	}
-	ctx.EventManager().EmitEvents(sdk.Events{sdk.NewEvent(
-		types.EventTypeRequest,
-		sdk.NewAttribute(types.AttributeKeyID, fmt.Sprintf("%d", id)),
-	)})
+
 	return &sdk.Result{Events: ctx.EventManager().Events().ToABCIEvents()}, nil
 }
 
 func handleMsgReportData(ctx sdk.Context, k Keeper, m MsgReportData) (*sdk.Result, error) {
 	if !k.IsReporter(ctx, m.Validator, m.Reporter) {
-		return nil, sdkerrors.Wrapf(types.ErrReporterNotAuthorized,
-			"val: %s, addr: %s", m.Reporter.String(), m.Validator.String())
+		return nil, types.ErrReporterNotAuthorized
```

Comment:
I don't think info about val, addr is needed. It's right in the message already.